oasislmf/pytools/cdftocsv.py

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env python
+
+import sys
+import os
+from contextlib import ExitStack
+import argparse
+import logging
+import numpy as np
+
+from oasislmf.pytools.getmodel.manager import get_mean_damage_bins
+from oasislmf.pytools.gul.common import ProbMean, damagecdfrec
+
+logger = logging.getLogger(__name__)
+
+parser = argparse.ArgumentParser(
+    usage='use "%(prog)s --help" for more information',
+    formatter_class=argparse.RawTextHelpFormatter  # for multi-line help text
+)
+
+parser.add_argument('-i', '--file-in', help='input filename', action='store', type=str)
+parser.add_argument('-s', help='skip header (default: False).', default=False, action='store_true', dest='skip_header')
+parser.add_argument('--run-dir', help='path to the run directory', default='.')
+parser.add_argument('--logging-level',
+                    help='logging level (debug:10, info:20, warning:30, error:40, critical:50)', default=30, type=int)
+
+
+def run(run_dir, skip_header, file_in=None, file_out=None):
+    """Run cdftocsv command: convert the cdf output from getmodel into csv format.
+    Presently, the csv file is streamed to stdout.
+
+    Args:
+        run_dir ([str]): Path to the run directory.
+        skip_header ([bool]): If True, does not print the csv header.
+        file_in ([str], optional): If provided, it reads the cdf from that file. Defaults to None.
+        file_out ([str], optional): Placeholder for a future output file writer. Defaults to None.
+
+    Raises:
+        NotImplementedError: If file_out is not None.
+        ValueError: If the stream type is not 1.
+
+    """
+    with ExitStack() as stack:
+        if file_in is None:
+            streams_in = sys.stdin.buffer
+
+        else:
+            streams_in = stack.enter_context(open(file_in, 'rb'))
+
+        if file_out is None:
+            stream_out = sys.stdout
+        else:
+            # implement here a file-writer (e.g., csv writer)
+            raise NotImplementedError("file_out has not been implemented yet.")
+
+        # get damage bins from file
+        static_path = os.path.join(run_dir, 'static')
+        damage_bins = get_mean_damage_bins(static_path=static_path)
+
+        # maximum number of damage bins (individual items can have up to `total_bins` bins)
+        if damage_bins.shape[0] == 0:
+            total_bins = 1000
+        else:
+            total_bins = damage_bins.shape[0]
+
+        # determine stream type
+        stream_type_mv = memoryview(bytearray(4))
+        stream_type = np.ndarray(1, buffer=stream_type_mv, dtype='i4')
+        len_read = streams_in.readinto(stream_type_mv)
```

Comment:
I think you should do dirrectly
len_read = np.frombuffer(stream_in.read(4), dtype='i4')

Reply:
Good idea, I've implemented the suggestion. Actually if I understand correctly `np.frombuffer` returns the array of the read values, not `len_read` so I've updated it with:
```py
stream_type = np.frombuffer(streams_in.read(4), dtype='i4')
```